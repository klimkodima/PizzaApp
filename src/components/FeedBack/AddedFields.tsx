import React from 'react';
import FormControl from '@mui/material/FormControl';
import Typography from '@mui/material/Typography';
import OutlinedInput from '@mui/material/OutlinedInput';

import { FormField } from '../../types';

const AddedFields = ( { formFields, handleChange } : {formFields: any, handleChange: any}) => {

    if (!formFields) return null;
  
    return (
      <>
        {formFields.map((field:FormField) => {
          return (
            <FormControl key={field.name}>
              <Typography component="label" variant="overline" color="text.primary" sx={{ fontWeight: 'bold' }}>
                {field.name}
              </Typography>
              <OutlinedInput id={field.id} name={field.name}
                sx={{ backgroundColor: "white" }}
                value=''
                onChange={handleChange}
              />
            </FormControl>
          )
        })
        }
      </>
    )
  }

  export default  AddedFields;