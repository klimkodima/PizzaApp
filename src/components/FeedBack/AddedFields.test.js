import React from 'react';
import renderer from 'react-test-renderer';
import AddedFields from './AddedFields';

describe('<AddedFields/>', () => {

    test('should render AddedFields without props', () => {

        const fields1 = []

        const component = renderer.create(
            <AddedFields formFields={fields1} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should render AddedFields with props', () => {

        const fields1 = [
            {
                id: "1",
                type: "text",
                name: "1"
            }
        ]

        const component = renderer.create(
            <AddedFields formFields={fields1} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});