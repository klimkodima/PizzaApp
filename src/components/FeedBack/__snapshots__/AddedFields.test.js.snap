// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddedFields/> should render AddedFields with props 1`] = `
<div
  className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <label
    className="MuiTypography-root MuiTypography-overline css-1itsahr-MuiTypography-root"
  >
    1
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-g0yakt-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="1"
      name="1"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-nnbavb"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`<AddedFields/> should render AddedFields without props 1`] = `null`;
